aws/cloudformation/cloud_formation_stack.yml.erb

```
@@ -14,6 +22,19 @@ Parameters:
     Type: String
     Default: <%=rack_env?(:production) ? '\'\'' : stack_name %>
     Description: Subdomain name for corporate web site ("Pegasus").
+  HourOfCodeBaseDomainName:
+    Type: String
+    Default: hourofcode.com
+    Description: Hour Of Code base domain name.
+  HourOfCodeHostedZoneId:
+    Type: String
+    # Default to ID for `hourofcode.com`
+    Default: Z3TQLAV9VNIB0G
+    Description: Route 53 Hosted Zone ID for Hour of Code base domain name.
+  HourOfCodeSubdomainName:
+    Type: String
+    Default: <%=rack_env?(:production) ? '\'\'' : stack_name %>
+    Description: Subdomain name for Hour Of Code site.
```

Can I suggest adding an AllowedPattern and ConstraintDescription to eliminate the risk of clobbering production:

```suggestion
  HourOfCodeSubdomainName:
    Type: String
    Default: <%=rack_env?(:production) ? '\'\'' : stack_name %>
    Description: Subdomain name for Hour Of Code site.
    AllowedPattern: "^(?!www$).*"
    ConstraintDescription: GitHubBranch cannot be `www` as this may conflict with the production environment.
```